presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsManager.java

```diff
@@ -41,9 +51,32 @@ public void addExternalPlanStatisticsProviderFactory(ExternalPlanStatisticsProvi
         this.externalPlanStatisticsProviderFactory = Optional.of(externalPlanStatisticsProviderFactory);
     }
 
-    public HistoryBasedPlanStatisticsCalculator getHistoryBasedPlanStatisticsCalculator(StatsCalculator delegate)
+    public void setDelegateStatsCalculator(StatsCalculator statsCalculator)
     {
-        return new HistoryBasedPlanStatisticsCalculator(
-                externalPlanStatisticsProviderFactory.map(factory -> factory.getExternalPlanStatisticsProvider()), metadata, delegate);
+        if (delegate.isPresent()) {
+            throw new IllegalStateException("delegate stats calculator can only be set once");
+        }
+        this.delegate = Optional.of(requireNonNull(statsCalculator, "stats calculator is null"));
+    }
```

Comment:
From the dependency injection perspective, which module are you waiting for? Is it `delegate` or something else?